### PR TITLE
Remove dependency between workflow cms

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: WorkflowFunctionAdminBundle/Controller/Admin/WorkflowRightController.php [hunk 1]
```diff
@@ -49,13 +49,12 @@ protected function generateForm(Request $request, WorkflowRightInterface $workfl
     {
         $form = $this->createForm('workflow_right', $workflowRight, array('action' => $url));
         $form->handleRequest($request);
-        $this->handleForm(
+        if($this->handleForm(
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/Controller/Admin/WorkflowRightController.php]: psr2

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function postUserUpdate(WorkflowRightEvent $event)
+    {
+        $workfolwRight = $event->getWorkflowRight();
+        $user = $this->userRepository->find($workfolwRight->getUserId());
+
+        $newRoles = array();
+
+        $authorizations = $workfolwRight->getAuthorizations();
+        foreach($authorizations as $authorization) {
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            $workflowFunctions = $authorization->getWorkflowFunctions();
+            foreach($workflowFunctions as $workflowFunction) {
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                $roles = $workflowFunction->getRoles();
+                foreach($roles as $role) {
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

File: WorkflowFunctionAdminBundle/Resources/config/collector.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+parameters:
+    open_orchestra_backoffice.collector.role.workflow_role_in_group: false
```
Comment (nicolasThal) [inline WorkflowFunctionAdminBundle/Resources/config/collector.yml]: I would have rather set à compiler pass for this parameter

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php (hunk 4 above)
Comment (ngilain) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    $newRoles[] = $role->getName();
+                }
+            }
+        }
+        $newRoles = array_diff(array_unique($newRoles), $user->getRoles());
+        foreach($newRoles as $newRole) {
```
Comment (ngilain) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

Comment (alavieille) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: psr2

File: WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            $user->addRole($newRole);
+        }
+        $this->documentManager->flush();
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/EventSubscriber/AddWorkFlowFunctionSubscriber.php]: I think it's better to flush only the user

File: WorkflowFunctionAdminBundle/Resources/config/collector.yml (hunk 5 above)
Comment (alavieille) [inline WorkflowFunctionAdminBundle/Resources/config/collector.yml]: you can change this parameter in the extension of worflow bundle

File: WorkflowFunctionAdminBundle/Tests/EventSubscriber/AddWorkFlowFunctionSubscriberTest.php [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+     * @param array $newRoles
+     * @param array $expectingRoles
+     *
+     * @dataProvider getExistingAndNewRoles
+     */
+    public function testPostUserUpdate($existingRoles, $newRoles, $expectingRoles)
+    {
+        $roles = new ArrayCollection();
+
+        foreach($newRoles as $newRole) {
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/Tests/EventSubscriber/AddWorkFlowFunctionSubscriberTest.php]: psr2

File: WorkflowFunctionAdminBundle/Tests/EventSubscriber/AddWorkFlowFunctionSubscriberTest.php [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            Phake::when($role)->getName()->thenReturn($newRole);
+            $roles->add($role);
+        }
+
+        Phake::when($this->workflowFunction)->getRoles()->thenReturn($roles);
+        Phake::when($this->user)->getRoles()->thenReturn($existingRoles);
+
+        $this->subscriber->postUserUpdate($this->workflowRightEvent);
+
+        foreach($expectingRoles as $expectingRole) {
```
Comment (alavieille) [inline WorkflowFunctionAdminBundle/Tests/EventSubscriber/AddWorkFlowFunctionSubscriberTest.php]: psr2

**Comment by veolys-nanne** [comment]
done @nicolasThal , @itkg-alavieille , @itkg-ngilain

**Comment by alavieille** [comment]
:+1:

**Pull request merged**